src/PatientChecking/PatientChecking/Controllers/PatientController.cs

```diff
@@ -132,5 +141,89 @@ public IActionResult Detail(int patientId)
 
             return View(model);
         }
+
+        [HttpPost]
+        public IActionResult Update(PatientDetailViewModel model)
+        {
+            if (!ModelState.IsValid)
+            {
+                ModelState.AddModelError("", "Unable to save changes. Please try again");
+            }
+
+            var patientDetails = new PatientDetails
+            {
+                PatientId = model.PatientId,
+                FirstName = model.FirstName,
+                MiddleName = model.MiddleName,
+                LastName = model.LastName,
+                FullName = string.IsNullOrEmpty(model.MiddleName) ? model.FirstName + " " + model.LastName : model.FirstName + " " + model.MiddleName + " " + model.LastName,
+                Nationality = model.Nationality,
+                DoB = DateTime.Parse(model.DoB),
+                MaritalStatus = model.MaritalStatus == (int)PatientMaritalStatus.Married,
+                Gender = (PatientGender)model.Gender,
+                EthnicRace = model.Nationality == "Vietnamese" ? model.EthnicRace : null,
+                HomeTown = model.HomeTown,
+                BirthplaceCity = model.BirthplaceCity,
+                InsuranceNo = model.InsuranceNo,
+                IdcardNo = model.IdcardNo,
+                IssuedDate = !string.IsNullOrEmpty(model.IssuedDate) ? DateTime.Parse(model.IssuedDate) : null,
+                IssuedPlace = model.IssuedPlace
+            };
+
+            var result = _patientService.UpdatePatientDetail(patientDetails);
+
+            if (result > 0)
+            {
+                _notyf.Success("Update patient detail successfully!");
+            }
+            else
+            {
+                _notyf.Error("Update patient detail Failed!");
+            }
+
+            return RedirectToAction("Detail", new { patientId = patientDetails.PatientId });
+        }
+
+        [HttpPost]
+        public IActionResult UploadImage(int patientId, IFormFile formFile)
+        {
+            try
+            {
+                if (formFile != null)
+                {
+                    var fileExtension = Path.GetExtension(formFile.FileName).ToLower();
+
+                    if (fileExtension != ".jpg" && fileExtension != ".png" && fileExtension != "jpeg")
```

Comment:
Please move the logic of uploading image into a `ImageService`

Reply:
All the logic of uploading image has been moved into ImageService. Please review this @vietvo361